Fix a bunch of warnings across the workspace

aumetra opened the pull request:
Sorry, the warnings were just annoying me *so much* when having the project open.

webmaster128 reviewed (commented):
Fixes are fine, but why isn't the CI complaining about those things? Can we make our CI stricter first such that we have those things covered in the future by default?

aumetra commented:
> Fixes are fine, but why isn't the CI complaining about those things?

Not sure! My local VSCode config is latest nightly + `cargo clippy` as the check command.  
That's the things it has shown me..

Yeah, I'll have a look at the CI config, that should be caught *at least* by the clippy step.

webmaster128 commented:
Mybe not for all of them, but some of those things are so obvious, they should be cought by Rust or Clippy from 2016. I think we are missing something there on the tooling side if things ...

aumetra commented:
After some googling, I think it's because we don't invoke the clippy tests with `cargo clippy --tests` since some of the basic warnings (like the unused static assertion traits) are gated behind `#[cfg(test)]` behind with the basic `cargo clippy` invocation can't look.  

And a bunch of the warnings I fixed were only present after I updated my nightly yesterday, so a bunch of those lints are probably *very* new.

webmaster128 commented:
But we always use `cargo clippy --all-targets`. Shouldn't that include the test code?

webmaster128 commented:
Could you have a look at https://github.com/CosmWasm/cosmwasm/pull/2067 first and merge that? Then we can see what's left here coming from nightly. Seems like something related to shadowing for things that are already imported.

aumetra commented:
> Shouldn't that include the test code?

I tested it locally and nightly clippy detects it just fine but stable clippy doesn't find a dead code warning for some reason

aumetra commented:
I'm wondering if it's just some new analysis on nightly, where it detects traits as unused if they aren't immediately used as a parameter (i.e. to bind a generic type parameter or to do dynamic dispatch).

I'll have a look into the Rust compiler PRs, I guess

aumetra commented:
Yep, seems like detecting the unused traits is part of https://github.com/rust-lang/rust/pull/118257  
So not on stable yet but part of beta

webmaster128 reviewed (approved):
Thanks
